subsys/bluetooth/host/gatt.c

```diff
@@ -878,8 +878,11 @@ static void sc_indicate(uint16_t start, uint16_t end);
 
 static void db_hash_process(struct k_work *work)
 {
+	bool store;
+
 #if defined(CONFIG_BT_SETTINGS)
-	if (atomic_test_and_clear_bit(gatt_sc.flags, DB_HASH_LOAD)) {
+	store = atomic_test_and_clear_bit(gatt_sc.flags, DB_HASH_LOAD);
```

Comment:
yes, in the case `store` = false. The prev. behavior was to always store, even if we hadn't actually loaded the hash from the NVS.

Reply:
Are you sure? It seems that the value of `store` is not used anywhere undef if and code under if ends with return. If BT_SETTINGS are disabled, the value of store is set to `true` anyway so behavior seems not changed.

Reply:
There is an additional `db_hash_store` under the `if` statement too

Reply:
this is inside the ifdef: 
![image](https://user-images.githubusercontent.com/68428270/218031755-e87bf970-9a08-4589-9d21-da911bdbd9f2.png)

Reply:
if the settings haven't yet been loaded, `store` will be false. previously we would then call `db_hash_gen` (at the end of this fn) with store=true, overwriting the value in the settings. 
Then when settings are loaded, the hash is recalculated, but it will match with the one in NVS (since it has just been overwritten).

Reply:
Ahh, you are right. There is a case where we do not enter `if` and it makes the difference. I initially overlooked it - sorry.

Reply:
so there was actually a bug here, I reworked it.

Reply:
Yes, thanks :)